File: backend/src/main/scala/domains/usergroup/objects/response/UserGroupSlugListResponse.scala
package domains.usergroup.objects.response

import domains.usergroup.objects.UserGroupSlug
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 用户所属用户组 slug 列表响应，供内部访问控制查询使用。 */
final case class UserGroupSlugListResponse(
  slugs: List[UserGroupSlug]
)

/** 提供用户组 slug 列表响应 JSON 编解码。 */
object UserGroupSlugListResponse:
  given Encoder[UserGroupSlugListResponse] = deriveEncoder[UserGroupSlugListResponse]
  given Decoder[UserGroupSlugListResponse] = deriveDecoder[UserGroupSlugListResponse]

File: frontend/src/objects/usergroup/response/UserGroupSlugListResponse.ts
import type { UserGroupSlug } from '@/objects/usergroup/UserGroupSlug'

/** 用户所属用户组 slug 列表响应；用于授权策略选择器等内部查询。 */
export type UserGroupSlugListResponse = {
  slugs: UserGroupSlug[]
}
